### PR TITLE
Updates fixes 2

### PR DESCRIPTION
**Pull request by zBritva**

### CODE REVIEW

File: src/app/views/panels/widgets/manager.tsx [hunk 1]
```diff
@@ -339,12 +339,20 @@ export class WidgetManager implements Prototypes.Controls.WidgetManager {
     return (
       <InputExpression
         defaultValue={this.getPropertyValue(property) as string}
-        validate={value =>
-          this.store.verifyUserExpressionWithTable(value, options.table)
-        }
+        validate={value => {
+          if (value.trim() !== "") {
```
Comment (stopyoukid) [inline src/app/views/panels/widgets/manager.tsx:343]: Can value be undefined/null here?

File: src/core/dataset/data_types.ts [hunk 2]
```diff
@@ -34,7 +34,12 @@ export let dataTypes: { [name in DataType]: DataTypeDescription } = {
     }
   },
   number: {
-    test: (x: string) => !isNaN(+x.replace(/\,/g, "")),
+    test: (x: string) => {
+      if (x === "null") {
```
Comment (stopyoukid) [inline src/core/dataset/data_types.ts:38]: Can `x` be null/undefined here?

File: src/app/views/panels/widgets/manager.tsx (hunk 1 above)
Comment (stopyoukid) [inline src/app/views/panels/widgets/manager.tsx:343]: Can value be undefined/null here?

File: src/core/prototypes/marks/data_axis.ts [hunk 3]
```diff
@@ -371,6 +372,26 @@ export class DataAxisClass extends MarkClass<
...
       groupBy: null // TODO: fixme
     };
+    let properties: Specification.Template.Property[] = [];
+    if (this.object.properties.axis) {
+      const axis = this.object.properties.axis;
+      properties = properties.concat(
+        buildAxisProperties(this.object as any, "axis")
+      );
+    }
+    if (this.object.properties.axis) {
```
Comment (stopyoukid) [inline src/core/prototypes/marks/data_axis.ts:382]: It seems like you can combine there two `if(this.object.properties.axis)` statements

File: src/core/dataset/data_types.ts (hunk 2 above)
Comment (zBritva) [reply src/core/dataset/data_types.ts:38]: No, the values come from string (it's part of string)so, only empty string suitable

File: src/core/prototypes/marks/data_axis.ts (hunk 3 above)
Comment (zBritva) [reply src/core/prototypes/marks/data_axis.ts]: Fixed

File: src/app/views/panels/widgets/manager.tsx (hunk 1 above)
Comment (zBritva) [reply src/app/views/panels/widgets/manager.tsx]: Add check

**Pull request merged**